networkx/algorithms/assortativity/tests/test_correlation.py

```diff
@@ -22,6 +22,10 @@ def test_degree_assortativity_directed(self):
         r = nx.degree_assortativity_coefficient(self.D)
         np.testing.assert_almost_equal(r, -0.57735, decimal=4)
 
+    def test_degree_assortativity_directed2(self):
```

Comment:
A bit more description to the test would be valuable - either renaming the test itself (which might get long) or by adding a docstring, e.g.

```suggestion
    def test_degree_assortativity_directed2(self):
        """Test degree assortativity for directed graph where set of
        in/out degree does not equal the total degree."""
```

Reply:
This just made me realize that the 2nd test (with D2 but using Pearson) is actually not really necessary because, given the implementation of the Pearson version, it cannot be subject to the issue encountered in the pure python function... should I remove it then or should I keep it and also add this docstring to it?

Reply:
I would say it would be good to keep the test if only because it checks that the Pearson implementation is done that way -- if it ever changes implementation in the future this test will check that feature.